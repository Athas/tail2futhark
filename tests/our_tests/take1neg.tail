let v0:<int>5 = take{[int],[1]}(-5,[1,2,3,4,5,6,7]) in
i2d(reduce{[int],[0]}(addi,0,v0))
