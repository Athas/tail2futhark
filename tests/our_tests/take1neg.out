fun real main() =
  let t_v0 = reshape((if (-5 <= 0) then --5 else -5),take1_int((-5 * 1),reshape(((size(0,[1,2,3,4,5,6,7]) * 1)),[1,2,3,4,5,6,7]))) in
  toReal(reduce(+,0,t_v0))